packages/notifier-seeder/src/models/catalogEventV2ConverterToV1.ts

```diff
@@ -0,0 +1,183 @@
+      { type: "EServiceDescriptorAdded" },
+      { type: "EServiceDescriptorPublished" },
+      { type: "EServiceDescriptorSuspended" },
+      { type: "EServiceDescriptorQuotasUpdated" },
+      (e) => {
+        const eserviceV1 = getEserviceV1(e);
+        const descriptorV1 = getDescriptorV1(eserviceV1, e.data.descriptorId);
+
+        return {
+          eserviceId: eserviceV1.id,
+          eserviceDescriptor: descriptorV1,
+        };
+      }
+    )
+    .with({ type: "EServiceDraftDescriptorDeleted" }, (e) => ({
+      eservice: getEserviceV1(e),
+      descriptorId: e.data.descriptorId,
+    }))
+    .with({ type: "EServiceDescriptorDocumentAdded" }, (e) => {
+      const eserviceV1 = getEserviceV1(e);
+      const descriptorV1 = getDescriptorV1(eserviceV1, e.data.descriptorId);
+      const documentV1 = getDocumentV1(descriptorV1, e.data.documentId);
+
+      return {
+        eserviceId: eserviceV1.id,
+        descriptorId: descriptorV1.id,
+        document: documentV1,
+        isInterface: false,
+        serverUrls: descriptorV1.serverUrls,
+      };
+    })
+    .with({ type: "EServiceDescriptorInterfaceAdded" }, (e) => {
+      const eserviceV1 = getEserviceV1(e);
+      const descriptorV1 = getDescriptorV1(eserviceV1, e.data.descriptorId);
+      const interfaceV1 = getInterfaceV1(descriptorV1, e.data.documentId);
+
+      return {
+        eserviceId: eserviceV1.id,
+        descriptorId: descriptorV1.id,
+        document: interfaceV1,
+        isInterface: true,
+        serverUrls: descriptorV1.serverUrls,
+      };
+    })
+    .with(
+      { type: "EServiceDescriptorDocumentDeleted" },
+      { type: "EServiceDescriptorInterfaceDeleted" },
+      (e) => {
+        if (!e.data.eservice) {
+          throw missingKafkaMessageDataError("eservice", e.type);
+        }
+
+        return {
+          eserviceId: e.data.eservice.id,
+          descriptorId: e.data.descriptorId,
+          documentId: e.data.documentId,
+        };
+      }
+    )
+    .with({ type: "EServiceDescriptorInterfaceUpdated" }, (e) => {
+      const eserviceV1 = getEserviceV1(e);
+      const descriptorV1 = getDescriptorV1(eserviceV1, e.data.descriptorId);
+      const interfaceV1 = getInterfaceV1(descriptorV1, e.data.documentId);
+
+      return {
+        eserviceId: eserviceV1.id,
+        descriptorId: descriptorV1.id,
+        documentId: interfaceV1.id,
+        updatedDocument: interfaceV1,
+        serverUrls: descriptorV1.serverUrls,
+      };
+    })
+    .with({ type: "EServiceDescriptorDocumentUpdated" }, (e) => {
+      const eserviceV1 = getEserviceV1(e);
+      const descriptorV1 = getDescriptorV1(eserviceV1, e.data.descriptorId);
+      const documentV1 = getDocumentV1(descriptorV1, e.data.documentId);
+
+      return {
+        eserviceId: eserviceV1.id,
+        descriptorId: descriptorV1.id,
+        documentId: documentV1.id,
+        updatedDocument: documentV1,
+        serverUrls: descriptorV1.serverUrls,
+      };
+    })
```

Comment:
these two have the same handler, can they be merged?

Reply:
Using different method `getDocumentV1` and `getInterfaceV1`